File: CodingSamples/Java/Language-Runtime/GenericMethodTest/demo1/SimpleStack.java
class SimpleStack<V> {

    private Node top;

    //inner member class
    class Node {

        V value;
        Node below;

        Node(V item) {
            value = item;
            below = top;
        }
    }

    public void push(V item) {
        top = new Node(item);
    }

    public V pop() {
        V item = top.value;
        top = top.below;
        return item;
    }

    public boolean empty() {
        return top == null;
    }

}

File: CodingSamples/DotNet/Language-Runtime/GenericTypeTest/DemoApp1/SimpleStack.cs
interface IStackReader<out E>
{
    E Pop();
    bool Empty();
}

interface IStackWriter<in E>
{
    void Push(E item);
}

class SimpleStack<E> : IStackReader<E>, IStackWriter<E>
{
    class Node
    {
        internal E Value;
        internal Node Below;
    }

    private Node top;

    public void Push(E item)
    {
        top = new Node { Value = item, Below = top };
    }

    public E Pop()
    {
        Node n = top;
        top = top.Below;
        return n.Value;
    }

    public bool Empty()
    {
        return top is null;
    }
}